src/pybind/mgr/cli_api/module.py

```diff
@@ -0,0 +1,105 @@
+import json
+import logging
+import threading
+import time
+from multiprocessing import Process
+from threading import Event
+
+from mgr_module import CLICommand, HandleCommandResult, MgrModule
+
+logger = logging.getLogger()
+
+
+
+class CLI(MgrModule):
+
+    NATIVE_OPTIONS = [
+        'mgr_tick_period',
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.run = True
+        self.event = Event()
+        self.mgr_tick_period = self.get_ceph_option("mgr_tick_period")
+        logger = self.log
+
+    @CLICommand('mgr api get')
+    def api_get(self, arg: str):
+        t1_start = time.time()
+        str_arg = self.get(arg)
+        t1_end = time.time()
+        time_final = (t1_end - t1_start)
+        return HandleCommandResult(0, json.dumps(str_arg), str(time_final))
+
+    @CLICommand('mgr api benchmark get')
+    def api_get_benchmark(self, arg: str, number_of_total_calls: int,
+                          number_of_parellel_calls: int):
+        benchmark_runner = ThreadedBenchmarkRunner(number_of_total_calls,
+                                                    number_of_parellel_calls)
+        benchmark_runner.start(self.get_method(arg))
+        benchmark_runner.join()
+        stats = benchmark_runner.get_stats()
+        return HandleCommandResult(0, json.dumps(stats), "")
+
+    def serve(self) -> None:
+        """
+        This method is called by the mgr when the module starts and can be
+        used for any background activity.
+        """
+        self.log.error("Starting")
+        while self.run:
+            # Do some useful background work here.
+
+            # Use mgr_tick_period (default: 2) here just to illustrate
+            # consuming native ceph options.  Any real background work
+            # would presumably have some more appropriate frequency.
+            sleep_interval = self.mgr_tick_period
+            self.log.error('Sleeping for %d seconds', sleep_interval)
+            self.event.wait(sleep_interval)
+            self.event.clear()
+
+    def get_method(self, arg: str):
+        def wrapper():
+            return self.get(arg)
+        return wrapper
```

Comment:
There's a Pythonic way to do this: [functools.partial](https://docs.python.org/3/library/functools.html#functools.partial)
```diff
- benchmark_runner.start(self.get_method(arg))
+ benchmark_runner.start(functools.partial(self.get, self, arg)) # you need to bind both self and the first arg
```